src/backends/tensorrt/plan_backend.cc

```diff
@@ -3117,14 +3165,23 @@ PlanBackend::Context::Run(
       // FIXME add correctness checking like below
       inference::DataType dt =
           ConvertTrtTypeToDataType(engine_->getBindingDataType(binding_index));
-      // Process the output tensors with the device memory address even if
-      // zero-copy may be used, so that the memory copies perform asynchronously
-      // and wait for model execution.
-      payload_->responder_->ProcessTensor(
-          name, io_binding_info.io_shape_mapping_.first, dt,
-          io_binding_info.io_shape_mapping_.second,
-          static_cast<const char*>(io_binding_info.device_buffer_),
-          TRITONSERVER_MEMORY_GPU, gpu_device_);
+      // Use pinned memory address if zero copy is supported, else use device
+      // memory address.
+      if (zero_copy_support) {
+        LOG_VERBOSE(1) << "Normal ProcessTensor";
+        payload_->responder_->ProcessTensor(
+            name, io_binding_info.io_shape_mapping_.first, dt,
+            io_binding_info.io_shape_mapping_.second,
+            static_cast<const char*>(io_binding_info.buffer_),
```

Comment:
Only this line differs? The just conditionalize this argument on zero_copy_support instead of the entire call